paymentsheet/src/main/java/com/stripe/android/link/gate/DefaultLinkGate.kt

```diff
@@ -3,14 +3,35 @@ package com.stripe.android.link.gate
 import com.stripe.android.core.utils.FeatureFlag
 import com.stripe.android.core.utils.FeatureFlags
 import com.stripe.android.link.LinkConfiguration
+import com.stripe.android.model.ElementsSession
+import com.stripe.android.model.StripeIntent
 import javax.inject.Inject
 
-internal class DefaultLinkGate @Inject constructor(
-    private val configuration: LinkConfiguration
+internal class DefaultLinkGate(
+    private val stripeIntent: StripeIntent,
+    private val useAttestationEndpointsForLink: Boolean,
+    private val suppressLink2faModal: Boolean,
+    private val disableRuxInFlowController: Boolean,
 ) : LinkGate {
+
+    @Inject
+    constructor(configuration: LinkConfiguration) : this(
+        stripeIntent = configuration.stripeIntent,
+        useAttestationEndpointsForLink = configuration.useAttestationEndpointsForLink,
+        suppressLink2faModal = configuration.suppress2faModal,
+        disableRuxInFlowController = configuration.disableRuxInFlowController,
+    )
+
+    constructor(elementsSession: ElementsSession) : this(
```

Comment:
👀 Added this constructor so we can determine attestation status before constructing `LinkConfiguration`.